packages/dynamic-import/dynamic-versions.js

```diff
@@ -30,3 +30,46 @@ exports.get = function (id) {
 
   return version;
 };
+
+function flattenModuleTree(tree) {
+  var parts = [""];
+  var result = Object.create(null);
+
+  function walk(t) {
+    if (t && typeof t === "object") {
+      Object.keys(t).forEach(function (key) {
+        parts.push(key);
+        walk(t[key]);
+        parts.pop();
+      });
+    } else if (typeof t === "string") {
+      result[parts.join("/")] = t;
+    }
+  }
+
+  walk(tree);
+
+  return result;
+}
+
+// If Package.appcache is loaded, preload additional modules after
+// the core bundle has been loaded
+if (Package.appcache) {
```

Comment:
Let's move the check for `Package.appcache` inside the `window.onload` callback, just in case the `appcache` package loads after `dynamic-import`.

Reply:
Also, in order to make he `window.onload` even compatible with all browsers, I would do something like [this](https://github.com/meteor/meteor/blob/87681c8f166641c6c3e34958032a5a070aa2d11a/packages/meteor/setimmediate.js#L110-L114):
```js
if (global.addEventListener) {
  global.addEventListener("load", handler, false);
} else {
  global.attachEvent("onload", handler);
}
```

Reply:
I'll make these changes - what version of IE does Meteor try to maintain support for? Any thought to using something like http://github.com/Raynos/DOM-shim ? It would allow us to remove forking code like this and still allow users who need IE8 and lower a way forward.

Also (this is more for my education) - is `global` a meteor variable?

Reply:
Also, do you think it'd be worth exporting `flattenModuleTree` - and have `client.js` (which now includes a copy of that function) consume that? Alternatively, I could more specifically tailor the copy I added to `dynamic-versions.js` and just leave it copied (or just leave it as is).

Reply:
> Also (this is more for my education) - is global a meteor variable?

https://github.com/tc39/proposal-global

Reply:
Yep, and the `meteor` package defines the `global` variable for all environments [here](https://github.com/meteor/meteor/blob/a7585da87c9e95b5413574f203a2a6adb1c628ff/packages/meteor/package.js#L22-L23).

I think this code is low-level enough that it's worth hand-coding cross-browser code, rather than bundling a library that we'd only use a part of. Likewise, I would support tailoring `flattenModuleTree` to the needs of this PR, especially if you see any opportunities to shorten/simplify the code.

Reply:
It's a minor change to flattenModuleTree - I could make it spit out the array I use directly, instead of converting that from the flat object it currently creates - it would use a bit less RAM, so I'll make that change.